presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -363,16 +363,24 @@ private static MemoryHistory getHistory()
     {
         MemoryHistory history;
```

Comment:
As commented below let's add a history file parameter, make this package-private, and mark this with `VisibleForTesting`.

Also we can early return to get rid of the `else` and the nestedness below:

```
@VisibleForTesting
static MemoryHistory getHistory(File historyFile)
{
    if (!historyFile.canWrite() || !historyFile.canRead()) {
        System.err.printf("WARNING: History file is not readable/writable: %s. " +
                        "History will not be available during this session.%n",
                historyFile.getAbsolutePath());
        MemoryHistory history = new MemoryHistory();
        history.setAutoTrim(true);
        return history;
    }

    MemoryHistory history;
    try {
        history = new FileHistory(historyFile);
        history.setMaxSize(10000);
    }
    catch (IOException e) {
        System.err.printf("WARNING: Failed to load history file (%s): %s. " +
                        "History will not be available during this session.%n",
                historyFile.getAbsolutePath(), e.getMessage());
        history = new MemoryHistory();
    }
    history.setAutoTrim(true);
    return history;
}
```

Reply:
That was my initial idea, but didn't see such examples in the code base so went hacking.